x-pack/libbeat/cmd/instance/beat.go

```diff
@@ -71,13 +71,29 @@ func NewBeatForReceiver(settings instance.Settings, receiverConfig map[string]an
 	}
 
 	cfg := (*config.C)(tmp)
-	if err := instance.InitPaths(cfg); err != nil {
-		return nil, fmt.Errorf("error initializing paths: %w", err)
+	if settings.Name == "filebeat" {
+		partialConfig := struct {
+			Path paths.Path `config:"path"`
+		}{}
+
+		if err := cfg.Unpack(&partialConfig); err != nil {
+			return nil, fmt.Errorf("error extracting default paths: %w", err)
+		}
+		p := paths.New()
+		if err := p.InitPaths(&partialConfig.Path); err != nil {
+			return nil, fmt.Errorf("error initializing default paths: %w", err)
+		}
+		b.Paths = p
+	} else {
+		if err := instance.InitPaths(cfg); err != nil {
```

Comment:
When this is eventually removed, we should reproduce the comment on why we do the partial paths unpack here

https://github.com/elastic/beats/blob/b6c79e4437dc98e5d038af7ad40a57c01c4b6318/libbeat/cmd/instance/beat.go#L1490-L1493

I had to go into this function definition to see why this was happening.

Reply:
I think it will get refactored so both beatreceivers and regular beats can just use a refactored version of `instace.InitPaths(cfg)` that returns a `*paths.Path`.